gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1665,7 +1665,93 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 				} else { // ASSIGN, or assignment operation (+=, -=, <<=, etc.)
 					// NOTE: Keep in sync with DEFINE above.
 					if len(n.Lhs) > len(n.Rhs) {
-						// check is done in assertCompatible
+						// check is done in assertCompatible where we also
+						// asserted we have at lease one element in Rhs
+						if cx, ok := n.Rhs[0].(*CallExpr); ok {
+							// we decompose the a,b = x(...) for named and unamed
+							// type value return in an assignments
+							// Call case: a, b = x(...)
+							ift := evalStaticTypeOf(store, last, cx.Func)
+							cft := getGnoFuncTypeOf(store, ift)
+							if len(n.Lhs) != len(cft.Results) {
```

Comment:
we already have this logic in type_check.go:
https://github.com/gnolang/gno/blob/b5560e20bb79ebff56eac956acddeca94e5f05f0/gnovm/pkg/gnolang/type_check.go#L721,

IMHO , we only need to keep one, and since this is related to type check, it would be more appropriate to put it in type_check.go, what do you think?